File: _helpers/libs/MyTeacher.Helper/Responses/BaseErrorResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace MyTeacher.Helper.Responses
{
    public class BaseErrorResponse
    {
        public BaseErrorResponse() {
            Payload = null;
            IsSuccessful = false;
            ErrorMessage = string.Empty;
            TraceId = string.Empty;
            ServerTime = 0;
            StatusCode = 0;
        }
        private BaseErrorResponse(string errorMessage, string traceId, HttpStatusCode statusCode)
        {
            Payload = null;
            IsSuccessful = false;
            ErrorMessage = errorMessage;
            TraceId = traceId;
            ServerTime = (int)(DateTime.UtcNow.Subtract(DateTime.UnixEpoch)).TotalSeconds;
            StatusCode = statusCode;
        }
        [JsonProperty]
        private object? Payload { get; }
        [JsonProperty]
        private bool IsSuccessful { get; }
        [JsonProperty]
        private string ErrorMessage { get; }
        [JsonProperty]
        private string TraceId { get; }
        [JsonProperty]
        private int ServerTime { get; }
        [JsonProperty]
        private HttpStatusCode StatusCode { get; }
        public static BaseErrorResponse BuildBaseErrorResponse(string errorMessage, string traceId, HttpStatusCode statusCode)
        {
            return new BaseErrorResponse(errorMessage, traceId, statusCode);
        }
    }
}

File: _client/MyTeacherClient/src/app/Helpers/BaseResponse/BaseErrorResponse.ts
export class BaseErrorResponse<T>{
    constructor(){
        this.error = new BaseErrorResponseModel<T>()
        this.message = ""
        this.name = ""
        this.ok = false
        this.status = 0
        this.statusText = ""
        this.url = ""
    }
    error: BaseErrorResponseModel<T>
    message: string
    name: string
    ok: boolean
    status: number
    statusText: string
    url: string
}
class BaseErrorResponseModel<T>{
    constructor(){
        this.isSuccessful = true
        this.serverTime = 0
        this.statusCode = 0
        this.traceId = ""
    }
    errorMessage?: string
    isSuccessful: boolean
    payload?: T
    serverTime: number
    statusCode: number
    traceId: string
}
export default BaseErrorResponse